korlibs-string/src@apple/korlibs/io/lang/NSDataExt.kt

```diff
@@ -0,0 +1,34 @@
+package korlibs.io.lang
+
+import kotlinx.cinterop.*
+import platform.CoreFoundation.CFRelease
+import platform.CoreFoundation.CFStringRef
+import platform.Foundation.*
+import platform.posix.memcpy
+
+@OptIn(ExperimentalForeignApi::class, UnsafeNumber::class)
+internal fun NSData.toByteArray(): ByteArray {
+    val result = ByteArray(length.toInt())
+    if (result.isEmpty()) return result
+
+    result.usePinned {
+        memcpy(it.addressOf(0), bytes, length)
+    }
+
+    return result
+}
+
+@OptIn(ExperimentalForeignApi::class, UnsafeNumber::class)
+internal fun ByteArray.toNSData(): NSData = NSMutableData().apply {
+    if (isEmpty()) return@apply
+    this@toNSData.usePinned {
+        appendBytes(it.addressOf(0), size.convert())
+    }
+}
+
+
+@OptIn(ExperimentalForeignApi::class)
+inline fun <T> String.useCFStringRef(block: (CFStringRef) -> T): T {
+    val ref = CFBridgingRetain(this as NSString) as CFStringRef
+    return block(ref).also { CFRelease(ref) }
+}
```

Comment:
Maybe internal to be consistent, or make public the other two

Reply:
@soywiz We can make them public and `NSData.toByteArray` is also used in korlibs-image. Since that is a duplicate, can we move this extension file to a separate module and include it in both modules?

Reply:
@itboy87 Yes of course. Feel free to move it to a common place.
I would add it to either `korlibs-platform` or `korlibs-memory`.